box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFile.java

```diff
@@ -1206,4 +1219,443 @@ public void setRequestedPage(int page) {
             mRequestPage = page;
         }
     }
+    /**
+     * Request for creating a chunked upload session for a new file
+     */
+    public static class CreateUploadSession extends BoxRequest<BoxUploadSession, CreateUploadSession> {
+        private static final long serialVersionUID = 8145675031279971502L;
+
+        private String mDestinationFolderId;
+        private String mFileName;
+        private long mFileSize;
+        private FileInputStream mFileInputStream;
+
+        /**
+         * Creates an upload file session  with the default parameters
+         *
+         * @param file file to upload
+         * @param destinationFolderId   id of the parent folder for the new file
+         * @param requestUrl    URL of the upload file endpoint
+         * @param session   the authenticated session that will be used to make the request with
+         */
+        public CreateUploadSession(File file, String destinationFolderId, String requestUrl, BoxSession session)
+                throws FileNotFoundException {
+            super(BoxUploadSession.class, requestUrl, session);
+            mRequestUrlString = requestUrl;
+            mRequestMethod = Methods.POST;
+            mFileName = file.getName();
+            mFileSize = file.length();
+            mFileInputStream = new FileInputStream(file);
+            mDestinationFolderId = destinationFolderId;
+            mBodyMap.put("folder_id", destinationFolderId);
+            mBodyMap.put("file_size", mFileSize);
+            mBodyMap.put("file_name", mFileName);
+        }
+
+        @Override
+        protected void onSendCompleted(BoxResponse<BoxUploadSession> response) throws BoxException {
+            if (response.isSuccess()) {
+                BoxUploadSession uploadSession = response.getResult();
+
+                try {
+                  computeSha1s(mFileInputStream, uploadSession, mFileSize);
+                } catch (NoSuchAlgorithmException e) {
+                    throw new BoxException("Can't compute sha1 for file",e);
+                } catch (IOException e) {
+                    throw new BoxException("Can't compute sha1 for file",e);
+                }
+            }
+            super.onSendCompleted(response);
+        }
+
+
+
+        //Pre-compute sha1s for sending in future calls and save them to BoxUploadSession
+        static void computeSha1s(FileInputStream fileInputStream, BoxUploadSession uploadSession,
+                                 long fileSize)
+                throws NoSuchAlgorithmException, IOException {
+            int totalParts = uploadSession.getTotalParts();
+            List<String> partSha1s = new ArrayList<>(totalParts);
+
+
+            byte[] partBuffer = new byte[uploadSession.getPartSize()];
```

Comment:
There is no need to declare this byte array here and inside the for loop. Lets keep the one inside the for loop